sandstone/client/sandstone/components/filetreedirective/tests/filetree.directive.test.js

```diff
@@ -1,208 +1,421 @@
+        });
+
+        it('describeSelection', function(){
+            var node = isolateScope.treeData.filetreeContents[0];
+            isolateScope.describeSelection(node, true);
+            expect(isolateScope.treeData.selectedNodes.length).toBe(1);
+            isolateScope.describeSelection(node, false);
+            expect(isolateScope.treeData.selectedNodes.length).toBe(0);
+        });
+
+        it('$rootScope.$on: filtree:created_file', function() {
+            spyOn(isolateScope, 'updateFiletree');
+            rootScope.$emit('filetree:created_file', {
+                filepath: '/tmp/some-file.txt'
+            });
+            expect(isolateScope.updateFiletree).toHaveBeenCalled();
+        });
+        it('$rootScope.$on: filtree:deleted_file', function() {
+            spyOn(isolateScope, 'updateFiletree');
+            rootScope.$emit('filetree:deleted_file', {
+                filepath: '/tmp/some-file.txt'
+            });
+            expect(isolateScope.updateFiletree).toHaveBeenCalled();
+        });
+        it('$rootScope.$on: filetree:moved_file', function() {
+            spyOn(isolateScope, 'updateFiletree');
+            rootScope.$emit('filetree:moved_file', {
+                src_filepath: '/home/user/somefile',
+                dest_filepath: '/home/user/some_new_file'
+            });
+            expect(isolateScope.updateFiletree).toHaveBeenCalled();
+        });
+      });
+      describe('leafLevel: dir', function() {
+          beforeEach(function() {
+              scope.sd = {
+                  noSelections: true,
+                  multipleSelections: false,
+                  dirSelection: false
+              };
+              var el = angular.element('<div sandstone-filetree tree-data="ctrl.treeData" leaf-level="dir" selection-desc="sd"></div>');
+              element = $compile(el)(scope);
+              scope.$digest();
+
+              // Get isolate scope
+              isolateScope = element.isolateScope();
+          });
+          it('initializeFiletree', function() {
+              // Create spies
+              spyOn(isolateScope, 'updateFiletree');
+              spyOn(isolateScope, 'deletedFile');
+              spyOn(isolateScope, 'pastedFiles');
+
+              expect(isolateScope.leafLevel).toBe("dir");
+              expect(isolateScope.selectionDesc.noSelections).toBeTruthy();
+              expect(isolateScope.selectionDesc.multipleSelections).not.toBeTruthy();
+              expect(isolateScope.selectionDesc.dirSelected).not.toBeTruthy();
+              expect(isolateScope.treeData.filetreeContents.length).toBe(2);
+          });
+          it('getDirContents', function(){
+              isolateScope.getDirContents(isolateScope.treeData.filetreeContents[0], true);
+              // First directory should have 4 files
+              expect(isolateScope.treeData.filetreeContents[0].children.length).toBe(4);
+          });
+      });
+      describe('template', function() {
```

Comment:
Description should be `directive`, as these are integration tests and cover much more than just the template.